packages/shared/src/utils/buildWorkflowPreferences.ts

```diff
@@ -42,3 +43,26 @@ export const buildWorkflowPreferences = (
     },
   };
 };
+
+/**
+ * Given a `critical` flag and a `IPreferenceChannels` object, build a `WorkflowPreferences` object
+ *
+ * @deprecated use `buildWorkflowPreferences` instead
+ */
+export const buildWorkflowPreferencesFromPreferenceChannels = (
+  critical: boolean = false,
+  preferenceChannels: IPreferenceChannels = {}
+): WorkflowPreferences => {
+  return buildWorkflowPreferences({
+    all: { enabled: true, readOnly: critical },
+    channels: Object.entries(preferenceChannels).reduce(
+      (output, [channel, value]) => ({
+        ...output,
+        [channel as ChannelTypeEnum]: {
+          enabled: value,
+        },
+      }),
+      {} as WorkflowPreferences['channels']
+    ),
+  });
+};
```

Comment:
Helper function for V1 workflow preferences.

Reply:
Will this be removed after the v1->v2 preference data migration?

Reply:
Yes, exactly. We will need to add a long-lived branched that folks can checkout to perform the migration step with the legacy repositories accessible, whilst our main branch will have it removed.